### PR TITLE
Adding LastInsertId for Go

### PR DESCRIPTION
**Pull request by Xeoncross**
#1459 adds support for [sql.Result.LastInsertId()](https://pkg.go.dev/database/sql#Result)

`:execlastinsertid` is calls `.LastInsertId()` rather than `.RowsAffected()` with `:execrows`. 

This will avoid the sql.Result database package leaking across the boundary of the pkg.Querier interface generated by sqlc when using MySQL & MariaDB

### CODE REVIEW

File: internal/codegen/golang/templates/pgx/queryCode.tmpl [hunk 1]
```diff
@@ -103,6 +103,23 @@ func (q *Queries) {{.MethodName}}(ctx context.Context, {{.Arg.Pair}}) (int64, er
...
+func (q *Queries) {{.MethodName}}(ctx context.Context, db DBTX, {{.Arg.Pair}}) (int64, error) {
+	result, err := db.Exec(ctx, {{.ConstantName}}, {{.Arg.Params}})
+{{- else -}}
+func (q *Queries) {{.MethodName}}(ctx context.Context, {{.Arg.Pair}}) (int64, error) {
+	result, err := q.db.Exec(ctx, {{.ConstantName}}, {{.Arg.Params}})
+{{- end}}
+	if err != nil {
+		return 0, err
+	}
+	return result.LastInsertId(), nil
```
Comment (Jille) [inline internal/codegen/golang/templates/pgx/queryCode.tmpl:119]: pgx doesn't support driver.Result.LastInsertId. People usually do "INSERT INTO ... RETURNING id" with :one when they want the sequence number from Postgres.

You should probably return an error when people try to use :execinsertid with Postgres.

**Review by kyleconroy — COMMENTED** [review]
> This will avoid the sql.Result database package leaking across the boundary of the pkg.Querier interface generated by sqlc when using MySQL & MariaDB

I'd rather leak the sql.Result than add another command. The reason is that we already support `:execresult`. I think adding `:execrows` was a mistake, so I don't want to make the same mistake twice.

**Comment by kyleconroy** [comment]
> Can you share more about why :execrows was a mistake?

In this case, you're right that it's not much more work to support. Also, there are only two methods on sql.Result, so we aren't going to have to add any more in the future.

File: internal/codegen/golang/templates/pgx/queryCode.tmpl (hunk 1 above)
Comment (kyleconroy) [inline internal/codegen/golang/templates/pgx/queryCode.tmpl:119]: You want `return result.LastInsertId()` here.

Comment (kyleconroy) [reply internal/codegen/golang/templates/pgx/queryCode.tmpl:119]: pgx also doesn't support the same signature for `RowsAffected()`. We'll need to address this is a different PR.

**Review by kyleconroy — CHANGES_REQUESTED** [review]
Since `RowsAffected()` maps to `:execrows`, let's map `LastInsertId()` to`:execlastid`.

Also, please add tests similar to [exec_rows](https://github.com/kyleconroy/sqlc/tree/main/internal/endtoend/testdata/exec_rows) and [exec_result](https://github.com/kyleconroy/sqlc/tree/main/internal/endtoend/testdata/exec_result).

**Comment by Xeoncross** [comment]
I tried to add the python generation example, but when I changed into the `internal/endtoend/testdata/exec_lastid/python_postgresql` folder and run it against my local sqlc it complained: 

```
$ ../../../../../sqlc2 generate
> error parsing sqlc.json: unknown target langauge "python"
```

I was using the same code from the [exec_rows](https://github.com/Xeoncross/sqlc/blob/feature-lastinsertid/internal/endtoend/testdata/exec_rows/python_postgresql/sqlc.json#L10) example.

I also wasn't sure what the PGX (go_postgresql_pgx) should be since this isn't something for PostgreSQL - so I left it.

**Comment by kyleconroy** [comment]
It's okay that you weren't able to add the Python support, can do that in a subsequent PR.

You're also right that we don't need to add support for pgx, as it's Result type does not have this method.

**Review by kyleconroy — APPROVED** [review]

**Pull request merged**